dogechain-lab/dogechain
# WIP: New pre-deploy contracts

# Description

1. This PR import two new pre-deploy contracts, `Bridge` and `Vault`.
Bridge contract is used for dogecoin blockchain crossing.
Vault contract is used for system reward collection and community profit sharing (T.B.D.).

2. Add an owner to `Staking` contract, who in charge of validator authorizations.
The owner would return rights to the community in the near future.

# Changes include


# Breaking changes

Different genesis means re-deployment of network or hard fork. Re-deploy network would be a better choice for now. 

# Checklist


## Testing


### Manual tests

1. Set up a local network of 4 nodes, which uses a Genesis sharing the same owner of all system contracts.
2. Try deposit, withdraw token from `Bridge` contract.
3. Check out the `ValidatorSet` on `Staking` contract.
4. Add and remove validator, using the owner authorization.
